src/sql/magic.py

```diff
@@ -639,6 +669,79 @@ def _persist_dataframe(
             table_name=table_name, data_frame=frame, if_exists=if_exists, index=index
         )
 
+    def get_query_type(self, command: SQLCommand, original: bool):
```

Comment:
there's no need for this function to be an instance method, it can be a separate function. so make it an independent function and add some unit tests

Reply:
Done